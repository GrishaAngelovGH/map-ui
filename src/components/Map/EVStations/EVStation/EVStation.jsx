const EVStation = ({ name, location, connector, distance, time }) => (
  <div className='row bg-white rounded p-3 m-5'>
    <div className='col-md-12'>
      <h3 className='text-center'>{name}</h3>
      <p className='text-center'>{location}</p>
      <p className='text-center'>Connector: {connector}</p>

      <div className='row justify-content-center'>
        <div className='col-5 col-md-4'>
          <div className='d-flex align-items-center'>
            <i className='bi bi-geo-alt fs-3 mx-2'></i>
            <span className='fw-bold'>{distance}km</span>
          </div>
        </div>
        <div className='col-5 col-md-5'>
          <div className='d-flex align-items-center'>
            <i className='bi bi-clock fs-3 mx-2'></i>
            <span className='fw-bold'>{time} minutes</span>
          </div>
        </div>
      </div>
    </div>
  </div>
)

export default EVStation