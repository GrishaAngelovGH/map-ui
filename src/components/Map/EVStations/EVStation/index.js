import EVStation from './EVStation'

export default EVStation