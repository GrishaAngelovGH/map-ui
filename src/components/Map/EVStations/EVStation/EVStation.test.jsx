import { render } from '@testing-library/react'

import EVStation from './EVStation'

test('should render EVStation component', () => {
  const view = render(
    <EVStation
      name='Station #1'
      location='Location 1'
      connector='Type 2'
      distance={1.5}
      time={0.8}
    />
  )

  expect(view).toMatchSnapshot()
})