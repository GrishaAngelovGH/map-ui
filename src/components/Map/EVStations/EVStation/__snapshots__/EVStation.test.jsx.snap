// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should render EVStation component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="row bg-white rounded p-3 m-5"
      >
        <div
          class="col-md-12"
        >
          <h3
            class="text-center"
          >
            Station #1
          </h3>
          <p
            class="text-center"
          >
            Location 1
          </p>
          <p
            class="text-center"
          >
            Connector: 
            Type 2
          </p>
          <div
            class="row justify-content-center"
          >
            <div
              class="col-5 col-md-4"
            >
              <div
                class="d-flex align-items-center"
              >
                <i
                  class="bi bi-geo-alt fs-3 mx-2"
                />
                <span
                  class="fw-bold"
                >
                  1.5
                  km
                </span>
              </div>
            </div>
            <div
              class="col-5 col-md-5"
            >
              <div
                class="d-flex align-items-center"
              >
                <i
                  class="bi bi-clock fs-3 mx-2"
                />
                <span
                  class="fw-bold"
                >
                  0.8
                   minutes
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="row bg-white rounded p-3 m-5"
    >
      <div
        class="col-md-12"
      >
        <h3
          class="text-center"
        >
          Station #1
        </h3>
        <p
          class="text-center"
        >
          Location 1
        </p>
        <p
          class="text-center"
        >
          Connector: 
          Type 2
        </p>
        <div
          class="row justify-content-center"
        >
          <div
            class="col-5 col-md-4"
          >
            <div
              class="d-flex align-items-center"
            >
              <i
                class="bi bi-geo-alt fs-3 mx-2"
              />
              <span
                class="fw-bold"
              >
                1.5
                km
              </span>
            </div>
          </div>
          <div
            class="col-5 col-md-5"
          >
            <div
              class="d-flex align-items-center"
            >
              <i
                class="bi bi-clock fs-3 mx-2"
              />
              <span
                class="fw-bold"
              >
                0.8
                 minutes
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
